### PR TITLE
[WIP] JSON Techwebs & visual techweb display/manipulator

### PR DESCRIPTION
**Pull request by silicons**
- [x] Get rid of all hard references to nodes and designs outside of research subsystem so garbage collection works (Yay, **HUGE REFACTOR!**)
- [x] Code in JSON de/serialization procs for all techweb datums
- [x] ^ techweb design datums
- [x] ^ techweb node datums
- [x] Make a configuration format for appending, replacing, and removing hardcoded techweb datums
- [ ] Make subsystem load from configuration and patch hardcoded techwebs/nodes/designs as necessary
- [x] Hook code as necessary so all research machinery/objects update when techweb/nodes/designs are modified mid-round

- [ ] **In-game admin techweb manipulator, visualizer and editor supporting json import/export and option to save edits to techwebs subsystem to modify the techweb for that round, and a dry run mode for experimentation only, as well as a reset to default button, R&D console visual techweb display and interaction**

Note: Optimally the default techweb setup would still be hardcoded on github, as I very much doubt headmins would want to be constantly badgered with science mains/anti science brigaders wanting techwebs patched without going through github review. 😎

### CODE REVIEW

**Comment by Armhulen** [comment]
what a kevinz pr. i like it!

**Comment by iskyp** [comment]
i t b e g i n s

**Comment by AlexTheSysop42** [comment]
Suggestion: make it so that nodes are hidden if any of their prereqs are (and have it chain so hiding, say, weapon tech hides beam weaponry as well)

File: config/research/techwebs_config.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+##While this may look like a regular config file, it is not parsed the same as a regular config file. # still denotes comments.
```
Comment (MrStonedOne) [inline config/research/techwebs_config.txt]: Considering we have a whole system to configs to make them modular, why isn't this parsed using the build-in config system?

@Cyberboss set this man straight

**Review by MrsTonedOne — COMMENTED** [review]
This PR modifies the MC's core, what could possibly go wrong.

File: config/research/techwebs_config.txt (hunk 1 above)
Comment (kevinz000) [reply config/research/techwebs_config.txt]: @MrStonedOne the regular configs require individual config entry datums for something to work, the techwebs primary one is just a list of files to include, the other files are processed by the subsystem. 2/4 of the types of files will just be .jsons, one of them is basically al ist of jsons by what the json is supposed to do, and the fourth is this one.

**Comment by MrStonedOne** [comment]
Please self-close pr or fix merge conflicts within 72 hours to avoid maintainer close.

**Pull request closed without merging**